src/devices/bus/nes/nes_ines.hxx

```diff
@@ -483,6 +483,18 @@ void nes_cart_slot_device::call_load_ines()
 			else if (submapper == 3)
 				pcb_id = IREM_HOLYDIVR;
 		}
+		// iNES Mapper 108
+		else if (mapper == 108)
+		{
+			if (submapper == 1)
+				pcb_id = UNL_DH08;
+			else if (submapper == 2)
+				pcb_id = UNL_LH31;
+			else if (submapper == 3)
+				pcb_id = UNL_LH28_LH54;
+			else if (submapper == 4)
+				pcb_id = UNL_LE05;
+		}
```

Comment:
Previously you had a hack in the handlers to infer the actual board type from incidental details like ROM size.  Would it be worth putting that hack here to deal with classic iNES without submapper info?  Or is iNES 2.0 widespread enough that the hack isn’t necessary?

Reply:
TL;DR: Thanks. I was silly and turned my brain off. Yes, the same info (CHR size and PPU mirroring) is among the few reliable details in iNES headers. I've switched to similar tests as the previous hack and tested it against loose .nes files.

Loading outside of the SL for NES is a bit of a mess. We support old iNES headers, NES 2.0 headers partially, UNIF format, and it's all tied together with the SL PCB loading stuff in 3 ugly .hxx includes that appear to wrap a lot of old C. Further with the iNES/NES 2.0 loading at least, in order to be compatible with all the broken headers out there, there are multiple pathways for getting metadata: old header, NES 2.0 header, plus those values being overwritten against a CRC check in the legacy hash/nes.hsi.

The result here is that unfortunately I've had to add two similar yet not identical switch cases. How I arrived at the approach here: 1) Said CRCs are not in nes.hsi, so we don't rely on that. 2) It's easier to ignore NES 2.0 submapper info and just go on the other incidental characteristics, plus the way the very long call_load is written the existing switch for doing the "incidental" test is AFTER the submapper test. 3) The two near identical switch cases are needed in separate functions (call_load and get_default_card_software). I don't pretend to understand the broader emu/di* stuff, but locally in bus/nes it's way beyond the scope of this PR to clean up/refactor the redundancies that currently exist in these two functions.